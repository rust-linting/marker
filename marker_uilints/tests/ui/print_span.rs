macro_rules! magic_macro {
    () => {
        "*magic penguin noises*"
    };
}
use magic_macro;

fn try_something() -> Option<u32> {
    let _span_try = Some(21)?;
    None
}

fn main() {
    let _span_normal = 178;

    let _span_macro = magic_macro!();
}
