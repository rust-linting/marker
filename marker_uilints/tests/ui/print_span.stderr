warning: print span
 --> $DIR/print_span.rs:9:5
  |
9 |     let _span_try = Some(21)?;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: Debug: $DIR/print_span.rs:9:21 - 9:30
  = note: Snippet: Some(21)?
  = note: Source: File(
              FileInfo {
                  file: "$DIR/print_span.rs",
                  span_src: SpanSrcId(..),
              },
          )
  = note: `#[warn(marker::print_span_lint)]` on by default

warning: print span
  --> $DIR/print_span.rs:14:5
   |
14 |     let _span_normal = 178;
   |     ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Debug: $DIR/print_span.rs:14:24 - 14:27
   = note: Snippet: 178
   = note: Source: File(
               FileInfo {
                   file: "$DIR/print_span.rs",
                   span_src: SpanSrcId(..),
               },
           )

warning: print span
  --> $DIR/print_span.rs:16:5
   |
16 |     let _span_macro = magic_macro!();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: Debug: [Inside Macro] $DIR/print_span.rs:16:23 - 16:37
   = note: Snippet: "*magic penguin noises*"
   = note: Source: Macro(
               ExpnInfo {
                   _lifetime: PhantomData<&()>,
                   parent: ExpnId(..),
                   call_site: SpanId(..),
                   macro_id: MacroId(..),
               },
           )

warning: 3 warnings emitted

